### PR TITLE
Fix hash collision

### PR DESCRIPTION
**Pull request by SerodioJ**
- adds UUIDObject base class
- changes pipeline to use object's UUID as key when possible, to avoid Python's hash collisions

### CODE REVIEW

**Comment by SerodioJ** [comment]
model_2-2-old_2
model_3-2-old

Examples of pipelines where hash collision happened

**Comment by SerodioJ** [comment]
model_0-2-new_q
Expected behaviour

File: dasf/transforms/base.py [hunk 1]
```diff
@@ -18,8 +18,13 @@
 from dasf.utils.types import is_dask_gpu_dataframe
 from dasf.utils.funcs import block_chunk_reduce
 
+class UUIDObject:
+    def get_uuid(self):
```
Comment (jcfaracco) [inline dasf/transforms/base.py:22]: I like the idea, but maybe we should use something like PipelineObject that we can reuse the structure maybe.

Comment (jcfaracco) [reply dasf/transforms/base.py:22]: I'm gonna merge this PR and make modifications based on some discussions we had previously.

**Pull request merged**